projects/packages/jetpack-mu-wpcom/bin/sync-newspack-blocks.sh

```diff
@@ -154,4 +156,36 @@ pnpm --package=jscodeshift dlx jscodeshift -t ./bin/sync-newspack-blocks-formatt
 echo "Changing PHP textdomain to match jetpack-mu-wpcom..."
 ../../../vendor/bin/phpcbf --standard=./.phpcs.dir.xml --filter=../../../vendor/automattic/jetpack-phpcs-filter/src/PhpcsFilter.php --runtime-set jetpack-filter-no-ignore -q $TARGET
 
+# Add textdomain to block.json
+echo "Adding textdomain to all block.json files..."
+for block_json_file in "$TARGET"/blocks/*/block.json; do
+	TMPFILE=$(mktemp)
+	jq --tab '. += {"textdomain": "jetpack-mu-wpcom"}' "$block_json_file" > "$TMPFILE"
+	mv "$TMPFILE" "$block_json_file"
+done
+
+# Generate PHPCS config file.
+echo "Generating .phpcs.dir.xml..."
+cat > "$TARGET"/.phpcs.dir.xml <<EOF
+<?xml version="1.0"?>
+<ruleset>
+
+	<rule ref="MediaWiki.Usage.ForbiddenFunctions">
+		<exclude name="MediaWiki.Usage.ForbiddenFunctions.isset"/>
+	</rule>
+
+</ruleset>
+EOF
+
+echo "Updating Phan baseline..."
+jetpack phan --update-baseline packages/jetpack-mu-wpcom
+
+# Warn about the need to use ENT_COMPAT.
+echo "Please ensure htmlentities and html_entity_decode use 'ENT_COMPAT'!"
+ent_compat_needed=$(grep -rino 'html_entity_decode\|htmlentities' --include="$TARGET/*.php")
+if [[ -n $ent_compat_needed ]]; then
+	echo 'Detected the below instances:'
+	echo "$ent_compat_needed"
+fi
```

Comment:
Won't this be a bunch of noise, since it doesn't actually check if `ENT_COMPAT` is already in use?

Ideally we'd let `PHPCompatibility.ParameterValues.NewHTMLEntitiesFlagsDefault` detect it, but first they'd have to release a version of PHPCompatibility that contains that rule. 🙁

OTOH, will the `.phpcsignore` take care of ignoring that too during the CI run? Is that the only reason this is here?

Reply:
> a bunch of noise

As far as I can see, it's only ever been one line to change, so I don't think it's too much noise. It's in there because the README said it was important to check for, and I figured it'd be helpful to point out places to check vs. assume the user knows what to look for. 🤷

> will the `.phpcsignore` take care of ignoring that too during the CI run?

Yeah, the synced folder will be ignored by CI, so it's a manual fix we need to be aware of (though I'm open to alternatives).

That said, I wonder if this is an issue upstream too, at which point a patch there would remove the need to fix it here... 🤔

Reply:
Yeah, if it needs fixing, the fix really should be upstreamed.

The consequences of not fixing it are that behavior will differ depending on the PHP version, PHP <8.1 vs 8.1+.

Reply:
Put in a PR upstream: Automattic/newspack-blocks#2010

Once that's merged, I'll remove that task from the script/README. :^)

Reply:
Removed now instead of waiting: bc7ac12621